Sources/ComposableArchitecture/SharedState/PersistenceKey.swift

```diff
@@ -7,9 +7,9 @@
 ///
 /// See the article <doc:SharingState> for more information, in particular the
 /// <doc:SharingState#Custom-persistence> section.
-public protocol PersistenceReaderKey<Value> {
+public protocol PersistenceReaderKey<Value>: Sendable {
```

Comment:
Persistence keys need to be sendable because they are captured in some `LockIsolated`'s, which currently require `Sendable` values and `@Sendable` transaction closures. If we weaken that, then perhaps this can be weakened too?